roofit/roofit/src/RooLagrangianMorphFunc.cxx

```diff
@@ -0,0 +1,3522 @@
+  extractCouplings(decCouplings, this->_decCouplings);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// config setter for couplings
+void RooLagrangianMorphFunc::Config::setCouplings(
+    const RooAbsCollection &couplings) {
+  extractCouplings(couplings, this->_couplings);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// config setter for production and decay couplings
+void RooLagrangianMorphFunc::Config::setCouplings(
+    const RooAbsCollection &prodCouplings,
+    const RooAbsCollection &decCouplings) {
+  extractCouplings(prodCouplings, this->_prodCouplings);
+  extractCouplings(decCouplings, this->_decCouplings);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// config setter for input file name
+void RooLagrangianMorphFunc::Config::setFileName(const char *filename) {
+  this->_fileName = filename;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// config setter for input folders
+void RooLagrangianMorphFunc::Config::setFolders(const RooArgList &folderlist) {
+  for (auto itr : folderlist)
+    this->_folderlist.add(*itr);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// config setter for name of the the observable to be morphed
+void RooLagrangianMorphFunc::Config::setObservableName(const char *obsname) {
+  this->_obsName = obsname;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// config setter for name of the the observable to be morphed
+void RooLagrangianMorphFunc::Config::allowNegativeYields(
+    bool isallowNegativeYields) {
+  this->_allowNegativeYields = isallowNegativeYields;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// config setter for diagrams
+template <class T>
+void RooLagrangianMorphFunc::Config::setDiagrams(
+    const std::vector<std::vector<T>> &diagrams) {
+  for (size_t j = 0; j < diagrams.size(); ++j) {
+    std::vector<RooArgList *> vertices;
+    for (size_t i = 0; i < diagrams[j].size(); i++) {
+      vertices.push_back(new RooArgList());
+      vertices[i]->add(diagrams[j][i]);
+    }
+    this->_configDiagrams.push_back(vertices);
+  }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// config setter for vertices
+template <class T>
+void RooLagrangianMorphFunc::Config::setVertices(
+    const std::vector<T> &vertices) {
+  std::vector<std::vector<T>> diagrams;
+  diagrams.push_back(vertices);
+  this->setDiagrams(diagrams);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// append the parameter map with a parmeter set
+
+void RooLagrangianMorphFunc::Config::append(
+    RooLagrangianMorphFunc::ParamMap &map, const char *str,
+    RooLagrangianMorphFunc::ParamSet &set) {
+  map[str] = set;
+}
```

Comment:
```suggestion
//void RooLagrangianMorphFunc::Config::append(
//    RooLagrangianMorphFunc::ParamMap &map, const char *str,
//    RooLagrangianMorphFunc::ParamSet &set) {
//  map[str] = set;
//}
```
Here we have another `std::map` copying that causes trouble with clang on macOS. Can we comment that function out for this PR? It is not used. I suppose at a later point when you do a PR with your further developments you will need it, but at that point it will also become more evident how we can change the code to avoid the copying.